modules/python3qt/src/properties/pythoneditordockwidget.cpp

```diff
@@ -27,47 +27,20 @@
  *
  *********************************************************************************/
 
-#include <modules/qtwidgets/properties/syntaxhighlighter.h>
-#include <inviwo/core/common/inviwomodule.h>
-#include <inviwo/core/util/filesystem.h>
-#include <modules/qtwidgets/qtwidgetssettings.h>
+#include <modules/python3qt/properties/pythoneditordockwidget.h>
+
 #include <inviwo/core/common/inviwoapplication.h>
 
-#include <warn/push>
-#include <warn/ignore/all>
-#include <QTextDocument>
-#include <QTextBlock>
-#include <warn/pop>
+#include <modules/python3qt/pythonsyntaxhighlight.h>
 
 namespace inviwo {
 
-SyntaxHighligther::SyntaxHighligther(QTextDocument* parent) : QSyntaxHighlighter(parent) {}
-
-SyntaxHighligther::~SyntaxHighligther() = default;
-
-const QColor& SyntaxHighligther::getBackgroundColor() const { return backgroundColor_; }
-
-void SyntaxHighligther::highlightBlock(const QString& text) {
-    setFormat(0, text.size(), defaultFormat_);
-
-    for (auto it = formaters_.begin(); it != formaters_.end(); ++it) {
-        SyntaxFormater::Result res = (*it)->eval(text, previousBlockState());
-
-        for (size_t i = 0; i < res.start.size(); i++) {
-            setFormat(res.start[i], res.length[i], *res.format);
-            setCurrentBlockState(res.outgoingState);
-        }
-    }
-}
+PythonEditorDockWidget::PythonEditorDockWidget(Property* property)
+    : TextEditorDockWidget(property) {
 
-template <>
-void SyntaxHighligther::loadConfig<None>() {
-    auto ivec4toQtColor = [](const ivec4& i) { return QColor(i.r, i.g, i.b, i.a); };
-    auto settings = InviwoApplication::getPtr()->getSettingsByType<QtWidgetsSettings>();
-    QColor textColor = ivec4toQtColor(settings->pyTextColor_.get());
-    backgroundColor_ = ivec4toQtColor(settings->pyBGColor_.get());
-    defaultFormat_.setBackground(backgroundColor_);
-    defaultFormat_.setForeground(textColor);
+    auto app = util::getInviwoApplication(property);
+    callbacks_ = utilqt::setPythonSyntaxHighlight(getSyntaxHighligther(),
```

Comment:
A common dockwidget where the callbacks are sent instead?

Reply:
I needs to be its own class for the factories.